nodi_backend/apps/datasets/views.py

```diff
@@ -41,20 +41,19 @@ def retrieve(self, request, pk=None):
         serializer = DataSetSerializer(dataset)
         return JsonResponse(serializer.data)
 
-    @action(detail=False, methods=['post'])
     def create(self, request):
         """
         +POST+
         Creates a dataset
         """
         body = json.loads(request.body)
-        d = DataSet(name=body.name, email=body.email,
-                    submitted_at=body.submitted_at, approved=False, bucket='nodi-unapproved-datasets', key=body.key, approved_at=None)
+        d = DataSet(name=body['name'], email=body['email'],
+                    submitted_at=body['submitted_at'], approved=False, bucket='nodi-unapproved-datasets', key=body.key, approved_at=None)
         d.save()
-        m = MetaData(publish_date=body.metadata.publish_date, department_ownership=body.metadata.department_ownership,
-                     raw_source_link=body.metadata.raw_source_link, description=body.metadata.description)
+        m = MetaData(publish_date=body['metadata']['publish_date'], department_ownership=body['metadata']['department_ownership'],
+                     raw_source_link=body['metadata']['raw_source_link'], description=body['metadata']['description'], dataset=d)
         m.save()
-        return self.s3_upload_url(request)
+        return JsonResponse(d)
```

Comment:
One more thing - can you use the serializer like in the other requests above this one?
Once you do that and fix the merge conflicts this should be good to go.